File: BlackJackJava/BJ.java
import java.util.Scanner;

public class BJ{
    private Deck deck = new Deck();
    private Hand playerHand = new Hand();
    private Hand dealerHand = new Hand();
    private Bet playerBet;
    
    public BJ(int amount){
        playerBet = new Bet(amount);
    }
    public void startGame(int betAmount){
        playerBet.placeBet(betAmount);
        deck.shuffle();
        playerHand.addCard(deck.deal());
        dealerHand.addCard(deck.deal());
        playerHand.addCard(deck.deal());
        dealerHand.addCard(deck.deal());
        displayResult(false);
    }

    public void playerHit(){
        playerHand.addCard(deck.deal());
        displayResult(false);
        if(playerHand.calculateValue() > 21){
            endGame("Busted");
            playerBet.loseBet();
        }
    }
    
    public void playerStand(){
        dealerTurn();
        displayResult(true);
        checkGameStatus();
    }

    private void dealerTurn(){ 
        while(dealerHand.calculateValue() < 17){
            dealerHand.addCard(deck.deal());
        }
    }

    private void checkGameStatus(){ 
        int playerValue = playerHand.calculateValue();
        int dealerValue = dealerHand.calculateValue();
        
        if(playerValue > 21){
            playerBet.loseBet();
            endGame("Busted");
        }
        else if(dealerValue > 21){
            playerBet.winBet();
            endGame("Win");
        }
        else if(playerValue == dealerValue){
            playerBet.tieBet();
            endGame("Tie");
        }
        else if(playerValue > dealerValue){
            playerBet.winBet();
            endGame("Win");
        }
        else{
            playerBet.loseBet();
            endGame("Lost");
        }
    }

    private void displayResult(boolean showDealerHand){
        System.out.println("Player's hand:");
        System.out.println(playerHand);
        System.out.println("Player's hand value: " + playerHand.calculateValue());

        System.out.println("Dealer's hand:");
        if (showDealerHand) {
            System.out.println(dealerHand);
            System.out.println("Dealer's hand value: " + dealerHand.calculateValue());
        } else {
            System.out.println(dealerHand.getCard().get(0));
            int singleValue = dealerHand.calculateCardValue(dealerHand.getCard().get(0));
            System.out.println("Dealer's visible card value: " + singleValue);
        }
    }

    private void endGame(String result){
        System.out.println(result);
        System.out.println("Your value: " + playerHand.calculateValue()); 
        System.out.println("Dealer value: " + dealerHand.calculateValue()); 
        System.out.println("Your Balance: " + playerBet.getBalance());        
    }
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        BJ game = new BJ(1000); 

        while(true){
            System.out.println("Enter your bet: ");
            int betAmount = scanner.nextInt();
            game.startGame(betAmount);
            
            while(true){
                System.out.println("!Hit (1) or !Stand (2)");
                int action = scanner.nextInt();

                if(action == 1){
                    game.playerHit();
                    if(game.playerHand.calculateValue() > 21){
                        break;
                    }
                }
                else if(action == 2){
                    game.playerStand();
                    break;
                }
                else{
                    System.out.println("Invalid Entry");
                }
            }
            
            System.out.println("Go again? (yes/no)");
            String again = scanner.next();
            if(again.equalsIgnoreCase("no")){
                 break;
            } 
            game.playerHand.resetHand();
            game.dealerHand.resetHand();
            game.deck.reset();
            game = new BJ(game.playerBet.getBalance());  
        }
        scanner.close();
    }
}

File: BlackJavaScript/BJ.js
class BJ{
    constructor(amount){
        this.deck = new this.deck();
        this.playerHand = new Hand();
        this.dealerHand = new Hand();
        this.playerBet = new this.playerBet(amount);
    }
    startGame(betAmount){
        this.playerBet.placeBet(betAmount);
        this.deck.shuffle();
        this.playerHand.addCard(this.deck.deal());
        this.dealerHand.addCard(this.deck.deal());
        this.playerHand.addCard(this.deck.deal());
        this.dealerHand.addCard(this.deck.deal());
        this.displayResult(false);
    }
    playerHit(){
        this.playerHand.addCard(this.deck.deal());
        this.displayResult(false);
        if(this.playerHand.calculateValue() > 21){
            this.endGame("Busted");
            this.playerBet.lostBet();
        }
    }
    playerStand(){
        this.dealerTurn();
        this.displayResult(true);
        this.checkGameStatus();
    }
    dealerTurn(){
        while(this.dealerHand.calculateValue() < 17){
            this.dealerHand.addCard(this.deck.deal());
        }
    }
    checkGameStatus(){
        const playerValue = this.playerHand.calculateValue();
        const dealerValue = this.dealerHand.calculateValue();

        if(playerValue > 21){
            this.playerBet.lostBet();
            this.endGame("Busted");
        }
        else if(dealerValue > 21){
            this.playerBet.winBet();
            this.endGame("Win");
        }
        else if(playerValue == dealerValue){
            this.playerBet.tieBet();
            this.endGame("Tie");
        }
        else if(playerValue > dealerValue){
            this.playerBet.winBet();
            this.endGame("Win");
        }
        else{
            this.playerBet.lostBet();
            this.endGame("Lost");
        }
    }
    displayResult(showDealerHand){
        console.log("Player's hand:");
        console.log(this.playerHand.toString());
        console.log("Player's hand value: " + this.playerHand.calculateValue());

        console.log("Dealer's hand:");
        if(showDealerHand){
            console.log(this.dealerHand.toString());
            console.log("Dealer's hand value: "+ this.dealerHand.calculateValue());
        }
        else{
            console.log(this.dealerHand.getCard()[0].toString());
            const singleValue = this.dealerHand.calculateCardValue(this.dealerHand.getCardA()[0]);
            console.log("Dealer's visible card value: " + singleValue);
        }
    }
    endGame(result){
        console.log(result);
        console.log("Your value: " + this.playerHand.calculateValue());
        console.log("Dealer value: " + this.dealerHand.calculateValue());
        console.log("Your Balance: " + this.playerBet.getBalance());
    }
}
